### PR TITLE
Reader Tests Fixing

### PR DESCRIPTION
**Pull request by ralphdoe**
- Pattern and Format Schema Properties added.
- Schema components added from parameters too.
- Tests are now enabled.
    - MergedOperationTest
    - AnnotatedOperationMethodTest
    - OperationsWithLinksTest
- Some refactors

### CODE REVIEW

File: modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/resources/SimpleUserResource.java [hunk 1]
```diff
@@ -20,6 +20,10 @@
 import io.swagger.jaxrs2.resources.model.User;
 import io.swagger.oas.annotations.Operation;
 import io.swagger.oas.annotations.Parameter;
+import io.swagger.oas.annotations.media.Content;
+import io.swagger.oas.annotations.media.Schema;
+import io.swagger.oas.annotations.responses.ApiResponse;
```
Comment (frantuma) [inline modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/resources/SimpleUserResource.java]: remove unused import

File: modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/annotations/pathItems/OperationsWithLinksTest.java [hunk 2]
```diff
@@ -16,7 +16,7 @@
 import static org.testng.Assert.assertEquals;
 
 public class OperationsWithLinksTest extends AbstractAnnotationTest {
-    @Test(enabled = false, description = "Shows creating simple links")
+    @Test(description = "Shows creating simple links")
     public void createOperationWithLinks() {
```
Comment (frantuma) [inline modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/annotations/pathItems/OperationsWithLinksTest.java]: Test fails, user operation not resolved at all, should resolve with correct links

File: modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/annotations/pathItems/OperationsWithLinksTest.java [hunk 3]
```diff
@@ -106,26 +106,34 @@ public Address getAddress(@QueryParam("userId") String userId) {
         private String zip;
     }
 
-    @Test(enabled = false, description = "Shows creating simple links")
+    @Test(description = "Shows creating simple links")
     public void createOperationWithLinkReferences() {
```
Comment (frantuma) [inline modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/annotations/pathItems/OperationsWithLinksTest.java]: test fails, also should use link references and resolve correctly

File: modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/resources/SimpleUserResource.java (hunk 1 above)
Comment (ralphdoe) [inline modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/resources/SimpleUserResource.java]: Done.

File: modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/annotations/pathItems/OperationsWithLinksTest.java (hunk 2 above)
Comment (ralphdoe) [inline modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/annotations/pathItems/OperationsWithLinksTest.java]: Done, the used .class was wrong, I had to use ClassWithOperationAndLinkReferences.

File: modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/annotations/pathItems/OperationsWithLinksTest.java (hunk 3 above)
Comment (ralphdoe) [inline modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/annotations/pathItems/OperationsWithLinksTest.java]: Done, the used .class was wrong, I had to use ClassWithOperationAndLinkReferences.

File: modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/annotations/pathItems/OperationsWithLinksTest.java (hunk 2 above)
Comment (ralphdoe) [inline modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/annotations/pathItems/OperationsWithLinksTest.java]: I see the components are missing.

File: modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/annotations/pathItems/OperationsWithLinksTest.java (hunk 3 above)
Comment (ralphdoe) [inline modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/annotations/pathItems/OperationsWithLinksTest.java]: The components are missing, I have to analyze that.

**Comment by frantuma** [comment]
closing this as it's included in https://github.com/swagger-api/swagger-core/pull/2361

**Pull request closed without merging**